lib/cc/engine/analyzers/violation.rb

```diff
@@ -45,7 +64,9 @@ def sorted_hashes
         end
 
         def other_sexps
-          @other_locations ||= sorted_hashes.drop(1)
+          sorted = sorted_hashes
```

Comment:
I guess part of what feels very strange to me is that `Violation` has this internal logic for determining which member of a set is it's "primary" and which are the "other" instances. That feels more like it should be the responsibility of whatever calls `Violation.new`, and `Violation.new` should get e.g. `current_sexp` & `other_sexps`.